src/errors.ts

```diff
@@ -196,19 +224,24 @@ export class ThrottlingError extends ServiceError {
 }
 
 /**
- * A failure due to the timeout being exceeded, but the timeout
- * was set lower than the query's expected processing time.
- * This response is distinguished from a ServiceTimeoutException
- * in that a QueryTimeoutError shows Fauna behaving in an expected
- * manner.
+ * A failure due to the query timeout being exceeded.
+ *
+ * This error can have one of two sources:
+ *     1. Fauna is behaving expectedly, but the query timeout provided was too
+ *        aggressive and lower than the query's expected processing time.
+ *     2. Fauna was not available to service the request before the timeout was
+ *        reached.
+ *
+ * In either case, consider increasing the `query_timeout_ms` configuration for
+ * your client.
  */
 export class QueryTimeoutError extends ServiceError {
```

Comment:
I merged the ServiceTimeoutError class into QueryTimeoutError. We talk about "client timeouts" vs "query timeouts" so I think that's the appropriate name, and QueryTimeoutError already had optional stats field defined.